wagtail_newsletter/campaign_backends/mailchimp.py

```diff
@@ -1,32 +1,80 @@
-from typing import Any
+import logging
+
+from copy import copy
+from dataclasses import dataclass
+from typing import Any, NoReturn, Optional, cast
 
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
+from django.utils.functional import cached_property
 from mailchimp_marketing import Client
 from mailchimp_marketing.api_client import ApiClientError
 
 from ..audiences import (
     Audience,
     AudienceSegment,
 )
-from . import CampaignBackend
+from ..models import NewsletterRecipientsBase
+from . import Campaign, CampaignBackend, CampaignBackendError
+
+
+CAMPAIGN_STATUS_DRAFT = "save"
+
+logger = logging.getLogger(__name__)
+
+
+def log_and_raise(error: ApiClientError, message: str, **kwargs) -> NoReturn:
```

Comment:
these 2 functions seem to belong to the private API of the module - if that's the case, their name would better be prefixed with an underscore to make that clear?

(That's more subjective, but... I would also maybe move them to the bottom of the module, so that developers opening this module don't start with code that doesn't really matter to them when they will be working with this package, but see the important public classes of this module instead?)

Reply:
Done. I like this pattern and will probably use it in the future. Thanks!